tests/unit/test_column_noise.py

```diff
@@ -60,9 +68,37 @@ def test_generate_missing_data(string_series, default_configuration):
     assert (string_series[not_noised_idx] == noised_data[not_noised_idx]).all()
 
 
-@pytest.mark.skip(reason="TODO")
-def test_incorrect_selection():
-    pass
+def test_incorrect_selection(categorical_series, default_configuration):
+    config = default_configuration["decennial_census"]["state"]["incorrect_selection"]
+    noised_data = generate_incorrect_selections(
+        categorical_series, config, RANDOMNESS0, "test_incorrect_select"
+    )
+    noised_data_same_seed = generate_incorrect_selections(
+        categorical_series, config, RANDOMNESS0, "test_incorrect_select"
+    )
+    noised_data_different_seed = generate_incorrect_selections(
+        categorical_series, config, RANDOMNESS1, "test_incorrect_select"
+    )
+
+    # Confirm same randomness stream provides same results
+    assert (noised_data == noised_data_same_seed).all()
+
+    # Confirm different streams provide different results
+    assert (noised_data != noised_data_different_seed).any()
+
+    # Check for expected noise level
+    expected_noise = config["row_noise_level"]
+    # todo: Update when generate_incorrect_selection uses exclusive resampling
+    # Get real expected noise to account for possibility of noising with original value
+    # Here we have a a possibility of choosing any of the 50 states for our categorical series fixture
+    expected_noise = expected_noise * (1 - 1 / 50)
```

Comment:
I think we might need to implement missingness (ie "" due to the `generate_missing_data` function that will always be run before any other column noising function) to ensure that's being correctly handled. These noise calculations will then need to be updated to account for that like you did in the integration test.

Reply:
If I'm understanding this correctly, I like this approach. If our input categorical series should has missingness, we are effectively testing that the interaction between missing data and this noise function is correct automatically. We can then have a single test that checks that two arbitrary noise functions when run together affect rows independently rather than needing to test every permutation.

Reply:
I think I triggered something smarter than I intended. We should probably talk about this b/c I'm not understanding what you're proposing @rmudambi